core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/offer/ContractOfferResolverImpl.java

```diff
@@ -117,24 +118,23 @@ private Stream<ContractOffer.Builder> createContractOffers(ContractDefinition de
 
     @NotNull
     private ContractOffer.Builder createContractOffer(ContractDefinition definition, Policy policy, Asset asset) {
-
-        var contractEndTime = calculateContractEnd(definition);
+        var now = ZonedDateTime.ofInstant(clock.instant(), clock.getZone());
```

Comment:
There is no point in converting to `ZonedDateTime`, pls use either `clock.instant().toEpochMilli()` or `Instant.now(clock).toEpochMilli()`

Reply:
I'm using the `now` variable to set the `contractStart` at line 128, therefore I need `ZonedDateTime`

Reply:
Yes I'm aware, but what I meant was to do the conversion to `ZonedDateTime` as "late" as possible, basically when calling the `ContractOffer.Builder#contractStart()` method, so when #341 lands we have a small change surface.